cpp/mcap/include/mcap/reader.inl

```diff
@@ -1752,7 +1742,17 @@ void LinearMessageView::Iterator::operator++(int) {
 }
 
 bool operator==(const LinearMessageView::Iterator& a, const LinearMessageView::Iterator& b) {
-  return a.impl_ == b.impl_;
+  if (a.impl_ == nullptr || b.impl_ == nullptr) {
+    // special case for Iterator::end() == Iterator::end()
+    return a.impl_ == b.impl_;
+  }
+  if (!a.begun_ && !b.begun_) {
+    // special case for Iterator::begin() == Iterator::begin()
+    // comparing iterators to the beginning of the same view should return true.
+    return &(a.impl_->view_) == &(b.impl_->view_);
+  }
+  // all other cases return false.
+  return false;
```

Comment:
Does this mean an iterator does not `==` itself (not reflexive) at least in the case of `begin() != begin()`? That seems potentially problematic